QUICKSTART.md

```diff
@@ -6,194 +6,77 @@
-gcp_credentials_json["private_key"] = bytes([_a ^ _b for _a, _b in zip(base64.b64decode(gcp_credentials_json["private_key"]), b"quickstart-sv"*150)]).decode("utf-8")
-credentials = GCPPipelineCredentials.from_services_dict(gcp_credentials_json, schema_prefix)
-```
-
-## 5. Create a DLT pipeline
-
-a. Instantiate a pipeline
-```
-pipeline = Pipeline(schema_name)
-```
-
-b. Create the pipeline with your credentials
-```
-pipeline.create_pipeline(credentials)
-```
+1. Configure DLT
 
-## 6. Load the data from the JSON document
+2. Create a DLT pipeline
 
-a. Load JSON document into a dictionary
-```
-with open('data.json', 'r') as f:
-    data = json.load(f)
-```
+3. Load the data from the JSON document
 
-## 7. Pass the data to the DLT pipeline
+4. Pass the data to the DLT pipeline
 
-a. Extract the dictionary into a table
-```
-pipeline.extract(iter(data), table_name=parent_table)
-```
+5. Use DLT to load the data
 
-b. Unpack the pipeline into a relational structure
-```
-pipeline.unpack()
-```
+## Running the code
 
-c. Save schema to `schema.yml` file
-```
-schema = pipeline.get_default_schema()
-schema_yaml = schema.as_yaml(remove_default_hints=True)
-with open(schema_file_path, 'w') as f:
-    f.write(schema_yaml)
-```
-
-
-## 8. Use DLT to load the data
-
-a. Load
-```
-pipeline.load()
-```
-
-b. Make sure there are no errors
-```
-completed_loads = pipeline.list_completed_loads()
-# print(completed_loads)
-# now enumerate all complete loads if we have any failed packages
-# complete but failed job will not raise any exceptions
-for load_id in completed_loads:
-    print(f"Checking failed jobs in {load_id}")
-    for job, failed_message in pipeline.list_failed_jobs(load_id):
-        print(f"JOB: {job}\nMSG: {failed_message}")
-```
-
-c. Run the script:
+Run the script:
 ```
 python3 quickstart.py
 ```
 
-d. Inspect `schema.yml` that has been generated:
+Inspect `schema.yml` that has been generated:
 ```
 vim schema.yml
 ```
 
-## 9. Query the Google BigQuery table
-
-a. Run SQL queries
-```
-def run_query(query):
-    df = c._execute_sql(query)
-    print(query)
-    print(list(df))
-    print()
-
-with pipeline.sql_client() as c:
-
-    # Query table for parents
-    query = f"SELECT * FROM `{schema_prefix}_example.json_doc`"
-    run_query(query)
-
-    # Query table for children
-    query = f"SELECT * FROM `{schema_prefix}_example.json_doc__children` LIMIT 1000"
-    run_query(query)
-
-    # Join previous two queries via auto generated keys
-    query = f"""
-        select p.name, p.age, p.id as parent_id,
-            c.name as child_name, c.id as child_id, c._pos as child_order_in_list
-        from `{schema_prefix}_example.json_doc` as p
-        left join `{schema_prefix}_example.json_doc__children`  as c
-            on p._record_hash = c._parent_hash
-    """
-    run_query(query)
-```
-
-b. See results like the following
+See results of querying the Google BigQuery table
```

Comment:
Near `SQL result:` I'd explain that this is the result of joining two tables above on autogenerated keys: `p._record_hash = c._parent_hash`. otherwise it's hard to guess what is what. maybe leaving the SQL queries themselves makes sense here?

Reply:
Very much agree. I did that